store/sqlstore/session_store.go

```diff
@@ -236,3 +216,10 @@ func (me SqlSessionStore) AnalyticsSessionCount() store.StoreChannel {
 		}
 	})
 }
+
+func (me SqlSessionStore) Cleanup() {
+	l4g.Debug("Cleaning up session store.")
+	if _, err := me.GetMaster().Exec("DELETE FROM Sessions WHERE ExpiresAt != 0 AND :ExpiresAt > ExpiresAt", map[string]interface{}{"ExpiresAt": model.GetMillis()}); err != nil {
```

Comment:
Actually I don't think we need `ExpiresAt != 0`. If an entry gets into the DB with 0 expiry I would like it deleted. There are no Sessions with ExpiresAt = 0 on pre-release.

Reply:
How many sessions are there likely to be in a large database? Because with data retention, it was causing problems if we deleted too many rows in a single query, causing massive replica lag while the query was running. Wondering if the same might happen here?

Reply:
If we guess 3 sessions per user and we're at 30k users that's 90k rows to go through, which doesn't sound too terrible. That's using the force, though.

For data retention, did you do anything to solve this? For example splitting the query into multiple queries that used offsets and limits instead of doing it all at once

Reply:
Take a look here for how data retention does it: https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/post_store.go#L1116

The `LIMIT` is necessary to ensure the actual "deletion" is fast, and the `< EndTime` is necessary to avoid a full table scan.

That being said, 90k rows isn't anywhere near 14M rows so it's probably fine without. I'll let @csduarte comment if he disagrees, given the feedback on the data retention queries came from him too.

Reply:
This issue came from Shift because they have bots creating many sessions and their sessions table was hitting 3M+ with many of that being from a single bot user. It was mainly a problem because the delete was happening on every session save, and probably won't be a big deal if it's only run once a day but since it looks relatively simple to do what you did in data retention I think it's worth implementing that here just to be safe